recipe-server/normandy/recipes/api/fields.py

```diff
@@ -42,3 +42,11 @@ def get_url(self, obj, view_name, request, format):
             url = reverse(view_name, kwargs=kwargs, request=request, format=format)
             assert url[0] == '/'
             return settings.CDN_URL + url[1:]
+
+    def to_representation(self, value):
+        # HyperlinkedIdentityField demands a request when creating
+        # urls, but one is not always available. Instead of erroring
+        # out hard, fake it.
```

Comment:
When calling `Action#canonical_json()`, such as for signing, it isn't part of a request/response cycle, so there is no request.